### PR TITLE
Hcap 1337 region sites

### PR DESCRIPTION
**Pull request by fw-noel**
Add a query on the endpoint to be able to get regions, and separate the region specific query.

I'm sorry that I couldn't avoid a flag, but I am trying to limit it to an extent!

https://freshworks.atlassian.net/browse/HCAP-1337

### CODE REVIEW

File: server/routes/employer-sites.js [hunk 1]
```diff
@@ -87,8 +89,38 @@ router.get(
...
+      },
+      sites_accessed: result.map((site) => site.siteId),
+    });
+    return res.json({ data: result });
+  })
+);
+
+// Read: Get All User Sites
+router.get(
+  '/user',
```
Comment (arranfw) [inline server/routes/employer-sites.js:108]: Why not sites/region?

I guess that wouldn't let us use multiple sites but I don't think that's a real use case is it?

**Review by arranfw — DISMISSED** [review]
Left a suggestion for route structure but not a blocker

**Review by alva-fresh — APPROVED** [review]

File: server/routes/employer-sites.js (hunk 1 above)
Comment (fw-noel) [reply server/routes/employer-sites.js:108]: If we do, on the frontend we'll have to differentiate the user on the Sites component. As in

if user.isHA, request sites/region
else if user.isMoH, request sites

I'm leaning towards not in this case. Don't love the query flag, but there's a little annoyance no matter which way we try it

Comment (arranfw) [reply server/routes/employer-sites.js:108]: Ah fair enough 👍

File: server/routes/employer-sites.js [hunk 2]
```diff
@@ -111,9 +112,15 @@ router.get(
     routeRedirect({ redirect: '/api/v1/employer-sites/details', match: 'employer-sites-detail' }),
   ],
   asyncMiddleware(async (req, res) => {
+    const { regions } = req.query;
     const { hcapUserInfo: user } = req;
 
-    const result = await getSitesForUser(user);
+    let result;
+    if (regions && user.isHA && user.regions.length > 0) {
```
Comment (arranfw) [inline server/routes/employer-sites.js:119]: After looking at this again, I think I'd prefer having this logic on the FE, then we have a more understandable

```
if (HA)
  GET (/sites/region)
else 
  GET (/sites)
```
and we can more easily handle this authorization in middleware

Still just a suggestion and not a blocker

**Review by arranfw — APPROVED** [review]

File: server/routes/employer-sites.js (hunk 2 above)
Comment (taraepp) [reply server/routes/employer-sites.js:119]: Yeah I'd like it as just a separate endpoint and keep all the logic of which endpoint to hit on the FE.

Comment (fw-noel) [reply server/routes/employer-sites.js:119]: Fair enough

**Review by taraepp — APPROVED** [review]
👍

**Review by arranfw — APPROVED** [review]

**Pull request merged**